### DISCUSSION TITLE
Extend a model with its own table

### DISCUSSION THREAD

**Discussion by Falci** [sequelize/sequelize#16010] [Help & Questions] [answered]
Would it be possible to extend a model, in a way that the generic model has its own table and can have associations?

Example:

```
class ClassRoom extends Model {
  capacity: number
}

class VirtualClassRoom extends ClassRoom {
   url: string
}

class PhysicalClassRoom extends ClassRoom {
  address: string;
}

class Booking extends Model {
  classRoom: ClassRoom
}
```

I would expect this to generate 4 tables, one for each. `VirtualClassRoom` and `PhysicalClassRoom` could have the same PK as their parent

**Comment by ephys** [comment]
See https://github.com/sequelize/sequelize/issues/6502

This will be part of Sequelize 7, and is already possible in Sequelize 6 if you use sequelize-typescript

**Reply by Falci** [reply]
Sorry, I failed to mention that I'm using sequelize-typescript

When I tried to run the example above, I end up with 4 tables, but the capacity was duplicated in all the 3 tables.
Also, there's no relationship between the children and the parent class.

Am I missing some config?

**Reply by ephys** [reply, marked as answer]
Ah unfortunately we don't support relationship-based inheritance. We generate a complete table for each of them

relationship-based inheritance is a feature that may be added in a future update
